envoy/api/v2/cds.proto

```diff
@@ -356,6 +356,19 @@ message Cluster {
     // .. note::
     //   The specified percent will be truncated to the nearest 1%.
     core.Percent healthy_panic_threshold = 1;
+    // [#not-implemented-hide:]
+    // Configures percentage of requests that will be routed to same zone
+    // If not specified, the default is 100%.
+    // :ref:`runtime values <config_cluster_manager_cluster_runtime_zone_routing>`.
+    // :ref:`Zone aware routing support <arch_overview_load_balancing_zone_aware_routing>`.
+    core.Percent zone_aware_routing_enabled = 2;
+    // [#not-implemented-hide:]
+    // Configures minimum upstream cluster size required for zone aware routing
+    // If upstream cluster size is less than specified ,zone aware routing is not performed
+    // even configured . If not specified, the default is 6.
+    // :ref:`runtime values <config_cluster_manager_cluster_runtime_zone_routing>`.
+    // :ref:`Zone aware routing support <arch_overview_load_balancing_zone_aware_routing>`.
+    google.protobuf.UInt64Value zone_aware_routing_min_cluster_size = 3;
```

Comment:
Once we have a couple of foo_bar_xxx, we are likely to have more similar fields in the future. It is much better to create a message FooBar to hold them. If `FooBar foo_bar` is null, then it is naturally disabled.

For naming, please avoid postadjectives, such as "items_collected". For boolean fields, enable_xxx is more clear and

Reply:
+1, to `ZoneAwareRouting` submessage for these fields.